src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -0,0 +1,371 @@
+
+                throw error;
+            })
+            .then(payment => {
+                return this._createOrder(payment, payload.useStoreCredit, options);
+            });
+    }
+
+    createButton(): HTMLElement {
+        return this._googlePaymentsClient.createButton({
+            buttonColor: ButtonColor.default,
+            buttonType: ButtonType.short,
+            onClick: this._handleWalletButtonClick,
+        });
+    }
+
+    private _configureWallet(): Promise<void> {
+        if (!this._methodId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(this._methodId))
+            .then(state => {
+                const paymentMethod = state.paymentMethods.getPaymentMethod(this._methodId);
+                const storeConfig = state.config.getStoreConfig();
+                const checkout = state.checkout.getCheckout();
+                const hasShippingAddress = !!state.shippingAddress.getShippingAddress();
+
+                if (!paymentMethod) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                if (!storeConfig) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+                }
+
+                if (!checkout) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCheckout);
+                }
+
+                this._paymentMethod = paymentMethod;
+                const testMode = paymentMethod.config.testMode;
+
+                return Promise.all([
+                    this._googlePayScriptLoader.load(),
+                    this._googlePayInitializer.initialize(checkout, paymentMethod, hasShippingAddress),
+                ])
+                    .then(([googlePay, googlePayPaymentDataRequest]) => {
+                        this._googlePaymentsClient = this._getGooglePaymentsClient(googlePay, testMode);
+                        this._googlePaymentDataRequest = googlePayPaymentDataRequest;
+                    })
+                    .catch((error: Error) => {
+                        this._handleError(error);
+                    });
+            });
+    }
+
+    private _displayWallet(): Promise<InternalCheckoutSelectors> {
+        return new Promise((resolve, reject) => {
+            if (!this._paymentMethod) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            if (!this._googlePaymentsClient && !this._googlePaymentDataRequest) {
+                throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+            }
+
+            this._googlePaymentsClient.isReadyToPay({
+                allowedPaymentMethods: this._googlePaymentDataRequest.allowedPaymentMethods,
+            }).then( (response: GooglePayIsReadyToPayResponse) => {
```

Comment:
You can omit `GooglePayIsReadyToPayResponse` because when you define the interface for `GooglePaymentsClient`, you already specified that `isReadyToPay` will return `Promise<GooglePayIsReadyToPayResponse>`.